test(Manager Backup): add native/rclone backup benchmarking under read/write stress

yarongilor opened the pull request:
This PR replaces https://github.com/scylladb/scylla-cluster-tests/pull/10806

Benchmark native/rClone backup  under read/write stress. Test duration of backup to S3 and stress latencies.
Fixes: #10764

### Testing
<!-- Add links to Argus/Jenkins of test test done with this PR -->
<!-- This would help the reviewer to cross check what was tested, and and review the results as needed -->
- [200GB test](https://argus.scylladb.com/tests/scylla-cluster-tests/e548d47b-3b54-44f7-8bf9-5da1c6c45785)
- [ubuntu24-sanity-test](https://argus.scylladb.com/tests/scylla-cluster-tests/52c131c8-abbe-4685-a54b-dc087432e1e0) passed ok.
- [upgrade test](https://argus.scylladb.com/tests/scylla-cluster-tests/a4f8b713-4774-44ea-85dd-ff0893794e1f) passed ok.

### PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

pehala commented on sdcm/mgmt/argus_report.py:
```diff
@@ -0,0 +1,37 @@
+    BACKUP = 2
+
+
+def report_to_argus(monitors, test_config, report_type: ManagerReportType, data: dict, label: str):
```
We are have one such function, why do we need more of them?

pehala commented on sdcm/mgmt/common.py:
```diff
@@ -1,18 +1,22 @@
 from enum import Enum
 import logging
 import datetime
+
```
Please refrain from unneccessary formatting changes

pehala commented on sdcm/tester.py:
```diff
@@ -3339,7 +3339,14 @@ def check_latency_during_ops(self, hdr_tags: list[str]):
                                         _registry=self.events_processes_registry))
         with open(self.latency_results_file, encoding="utf-8") as file:
-            latency_results = json.load(file)
+            try:
```
Why? This is a pretty huge change

yarongilor replied:
this is a small change - just validate a file is opened and json successfully loaded before continuing.
i know it fails some times and it doesn't make sense to fail the whole test because of that.
it might be a good idea to further investigate and fix it in a followup PR.

pehala commented on test-cases/performance/perf-regression-latency-backup-nemesis.yaml:
```diff
@@ -0,0 +1,41 @@
+append_scylla_args: '--blocked-reactor-notify-ms 5 --abort-on-lsa-bad-alloc 1 --abort-on-seastar-bad-alloc --abort-on-internal-error 1 --abort-on-ebadf 1'
+backtrace_decoding: false
+
+store_perf_results: false
```
Do we need all of these otpions?

pehala reviewed (changes requested):
The commit structure does not make sense to me, I would redo it to better isolate changes

yarongilor commented:
> The commit structure does not make sense to me, I would redo it to better isolate changes

@pehala , can you please elaborate specific concerns? basically most of the commits code could be squashed as well.
are you looking for a specific change to isolate?

mikliapko commented:
@yarongilor Please, run some basic Manager tests to verify your refactoring changes to `mgmt_cli_test.py`.
You can pick up some tests from manager-master (https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/) - at least [ubuntu24-sanity-test](https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/job/ubuntu24-sanity-test/), [ubuntu24-upgrade-test](https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/job/ubuntu24-upgrade-test/).
I see no tests triggered and attached to this PR.

yarongilor commented on test-cases/performance/perf-regression-latency-backup-nemesis.yaml:
```diff
@@ -0,0 +1,41 @@
+                     "cassandra-stress write cl=ALL n=1500000000 -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native  -rate threads=650 -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=4500000001..6000000000" ]
+
+stress_cmd_m: "cassandra-stress mixed cl=QUORUM duration=720m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=144 -col 'size=FIXED(1024) n=FIXED(1)' -pop 'dist=gauss(1..6000000000,3000000000,900000000)'"
+instance_type_db: 'i4i.8xlarge'
```
The last commit from Ran was mistakenly removed. it's back now.
`test(Manager backup benchmark): Added test setups for 2TB and 200GB`

regevran commented on configurations/manager/200GB_backup_dataset.yaml:
```diff
@@ -0,0 +1,12 @@
+
+# fill the cluster with data
+prepare_write_cmd: [
+    "cassandra-stress write cl=ALL n=209715200 -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=650 -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=1..209715200",
```
`seq=1..209,715,200` is actually ~210GB on its own.
4 of the same range is ~850GB overall and not as the name of the file suggests.

yarongilor replied:
renamed to 850GB_backup_dataset.yaml

yarongilor commented on test-cases/performance/perf-regression-latency-backup-nemesis.yaml:
```diff
@@ -0,0 +1,41 @@
+n_monitor_nodes: 1
+region_name: us-east-1
+backup_bucket_location: 'manager-backup-tests-us-east-1'
+append_scylla_yaml:
```
fixed by:
```
+# Calculate stream_io_throughput_mb_per_sec for i4i.2xlarge:
+# It has Network bandwidth = 6.25 GiB/s
+# GiB/s to Mib/s
+# 6.25 GiB/s = 6250 Mib/s
+# Mib/s to MB/s (disk-wise)
+# Divide by 10 (not 8) → 6250 / 10 = 625 MB/s
+# Convert to "disk MB/s" using 1000/1024: 625 * 1000 / 1024 = ~610.35 MB/s
+# Take 75% of that: 0.75 * 610.35 = ~457.76 MB/s
+append_scylla_args: '--stream_io_throughput_mb_per_sec 458 --stream-io-throughput-mb-per-sec 600 --blocked-reactor-notify-ms 5 --abort-on-lsa-bad-alloc 1 --abort-on-seastar-bad-alloc --abort-on-internal-error 1 --abort-on-ebadf 1'
```

regevran replied:
> @regevran , this file is not found: https://github.com/scylladb/scylla-machine-image/tree/next/common/aws_net_params.json

Correct, it waits for https://github.com/scylladb/scylla-machine-image/issues/742 to be merged.

yarongilor commented on mgmt_cli_test.py:
```diff
@@ -1922,38 +1166,35 @@ def test_one_to_one_restore(self):
 
 class ManagerReportType(Enum):
     READ = 1
-    BACKUP = 2
+    WRITE = 2
+    BACKUP = 3
```
rebase issue.

yarongilor commented on jenkins-pipelines/manager/restore_benchmark/ubuntu22-manager-1tb-1t-3nodes.jenkinsfile:
```diff
@@ -7,7 +7,7 @@ managerPipeline(
     region: 'us-east-1',
     provision_type: 'on_demand',
-    test_name: 'mgmt_cli_test.ManagerRestoreBenchmarkTests.test_restore_benchmark',
+    test_name: 'scylla_manager_performance_test.ManagerRestoreBenchmarkTests.test_restore_benchmark',
```
rebase issue.

yarongilor commented on jenkins-pipelines/manager/ubuntu22-manager-backup-restore-custom-dataset.jenkinsfile:
```diff
@@ -6,7 +6,7 @@ def lib = library identifier: 'sct@snapshot', retriever: legacySCM(scm)
     backend: 'aws',
     region: 'us-east-1',
-    test_name: 'mgmt_cli_test.ManagerRestoreBenchmarkTests.test_restore_benchmark',
+    test_name: 'scylla_manager_performance_test.ManagerRestoreBenchmarkTests.test_restore_benchmark',
```
rebase issue.

yarongilor commented on configurations/manager/manager_native_backup.yaml:
fixed: configurations/manager/manager_native_backup_nemesis.yaml

yarongilor commented on configurations/manager/manager_rclone_backup.yaml:
fixed: configurations/manager/manager_rclone_backup_nemesis.yaml

yarongilor commented on sdcm/mgmt/argus_report.py:
```diff
@@ -0,0 +1,37 @@
+from sdcm.mgmt.common import get_backup_size
+
+
+class ManagerReportType(Enum):
+    READ = 1
+    BACKUP = 2
```
rebase issue

yarongilor commented on sdcm/mgmt/common.py:
```diff
@@ -180,3 +183,44 @@ class AgentBackupParameters(BaseModel):
+    NATIVE = "native"
+
+
+def manager_backup(mgr_cluster, locations, object_storage_upload_mode=None, timeout=7200):
```
moved to operations.py.
renamed to run_manager_backup.

yarongilor commented on jenkins-pipelines/manager/ubuntu22-manager-restore-data-outside-of-manager.jenkinsfile:
```diff
@@ -6,7 +6,7 @@ def lib = library identifier: 'sct@snapshot', retriever: legacySCM(scm)
     backend: 'aws',
     region: 'us-east-1',
-    test_name: 'mgmt_cli_test.ManagerRestoreBenchmarkTests.test_restore_data_without_manager',
+    test_name: 'scylla_manager_performance_test.ManagerRestoreBenchmarkTests.test_restore_data_without_manager',
```
rebase issue.

yarongilor commented on jenkins-pipelines/manager/manager-backup-native-backup.jenkinsfile:
```diff
@@ -0,0 +1,11 @@
+managerPipeline(
+    backend: 'aws',
+    region: 'us-east-1',
+    test_name: 'scylla_manager_performance_test.ManagerBackupRestoreConcurrentTests.test_native_backup_benchmark',
```
rebase issue.

yarongilor commented on configurations/manager/6TB_dataset.yaml:
file is removed.

fruch commented on sdcm/utils/decorators.py:
```diff
@@ -30,7 +30,6 @@
 from sdcm.sct_events.event_counter import EventCounterContextManager
 from sdcm.exceptions import UnsupportedNemesis
 from sdcm.sct_events.system import TestFrameworkEvent
-from sdcm.utils.cluster_tools import check_cluster_layout
```
This PR is creating the circular import by importing from sdcm.utils.common

The ParallelObject should be refactored into its own module

yarongilor commented:
> > > @yarongilor Please, run some basic Manager tests to verify your refactoring changes to `mgmt_cli_test.py`. You can pick up some tests from manager-master (https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/) - at least [ubuntu24-sanity-test](https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/job/ubuntu24-sanity-test/), [ubuntu24-upgrade-test](https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/job/ubuntu24-upgrade-test/). I see no tests triggered and attached to this PR.
> > 
> > 
> > running a test in: https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/job/ubuntu24-sanity-debug/3/
> > [ubuntu24-sanity-test](https://argus.scylladb.com/tests/scylla-cluster-tests/52c131c8-abbe-4685-a54b-dc087432e1e0) passed ok.
> > [upgrade test](https://argus.scylladb.com/tests/scylla-cluster-tests/a4f8b713-4774-44ea-85dd-ff0893794e1f) passed ok.
> > cc: @mikliapko
> 
> @yarongilor Please, take a look at **Collect log data** and **Send email with results** steps of jobs you triggered - they failed in both mentioned jobs (I'd suspect refactoring issue).

@mikliapko , i don't see any way these failures could ever be related to this PR. Let me know if you suspect anything specific.

pehala commented:
> @mikliapko , i don't see any way these failures could ever be related to this PR. Let me know if you suspect anything specific.
Try rebasing and rerunning if it fail again

mikliapko commented:
> > > > @yarongilor Please, run some basic Manager tests to verify your refactoring changes to `mgmt_cli_test.py`. You can pick up some tests from manager-master (https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/) - at least [ubuntu24-sanity-test](https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/job/ubuntu24-sanity-test/), [ubuntu24-upgrade-test](https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/job/ubuntu24-upgrade-test/). I see no tests triggered and attached to this PR.
> > > 
> > > 
> > > running a test in: https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/job/ubuntu24-sanity-debug/3/
> > > [ubuntu24-sanity-test](https://argus.scylladb.com/tests/scylla-cluster-tests/52c131c8-abbe-4685-a54b-dc087432e1e0) passed ok.
> > > [upgrade test](https://argus.scylladb.com/tests/scylla-cluster-tests/a4f8b713-4774-44ea-85dd-ff0893794e1f) passed ok.
> > > cc: @mikliapko
> > 
> > 
> > @yarongilor Please, take a look at **Collect log data** and **Send email with results** steps of jobs you triggered - they failed in both mentioned jobs (I'd suspect refactoring issue).
> 
> @mikliapko , i don't see any way these failures could ever be related to this PR. Let me know if you suspect anything specific.

Take a look at this test, the same scenario but for Ubuntu22 run last night from scylladb/master - it doesn't contain any failures in **Collect logs** or **Send email** steps.
https://jenkins.scylladb.com/job/manager-master/job/ubuntu22-sanity-test/745/

pehala commented on sdcm/nemesis.py:
```diff
@@ -6804,3 +6847,21 @@ class IsolateNodeWithIptableRuleNemesis(Nemesis):
+    supports_high_disk_utilization = False
+
+    def disrupt(self):
+        self.disrupt_manager_backup(object_storage_upload_mode=ObjectStorageUploadMode.RCLONE, label='rclone_backup')
```
this will never be executed

pehala commented on sdcm/nemesis.py:
```diff
@@ -6804,3 +6847,21 @@ class IsolateNodeWithIptableRuleNemesis(Nemesis):
+    supports_high_disk_utilization = False
+
+    def disrupt(self):
+        self.disrupt_manager_backup(object_storage_upload_mode=ObjectStorageUploadMode.NATIVE, label='native_backup')
```
this will never be executed, the nemesis will be run, `self.disrupt_manager_backup()` will be executed instead

fruch replied:
It would be selected but won't use the code mentioned above.

If there is a reason you do want this nemesis run in all cases, it should be documented within the code

And if it's needed to ignore by default it should be added to the complex nemesis list to be excluded explicitly,  now like that, since the code might be refact at some point and it would start running

yarongilor replied:
@pehala , fixed that in a new commit - added nemesis to excluded list.

yarongilor commented:
> > > > > @yarongilor Please, run some basic Manager tests to verify your refactoring changes to `mgmt_cli_test.py`. You can pick up some tests from manager-master (https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/) - at least [ubuntu24-sanity-test](https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/job/ubuntu24-sanity-test/), [ubuntu24-upgrade-test](https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/job/ubuntu24-upgrade-test/). I see no tests triggered and attached to this PR.
> > > > 
> > > > 
> > > > running a test in: https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/job/ubuntu24-sanity-debug/3/
> > > > [ubuntu24-sanity-test](https://argus.scylladb.com/tests/scylla-cluster-tests/52c131c8-abbe-4685-a54b-dc087432e1e0) passed ok.
> > > > [upgrade test](https://argus.scylladb.com/tests/scylla-cluster-tests/a4f8b713-4774-44ea-85dd-ff0893794e1f) passed ok.
> > > > cc: @mikliapko
> > > 
> > > 
> > > @yarongilor Please, take a look at **Collect log data** and **Send email with results** steps of jobs you triggered - they failed in both mentioned jobs (I'd suspect refactoring issue).
> > 
> > 
> > @mikliapko , i don't see any way these failures could ever be related to this PR. Let me know if you suspect anything specific.
> 
> Take a look at this test, the same scenario but for Ubuntu22 run last night from scylladb/master - it doesn't have any failures in **Collect logs** or **Send email** steps. https://jenkins.scylladb.com/job/manager-master/job/ubuntu22-sanity-test/745/

@mikliapko , again, these errors are NOT related to this PR.
i even confirmed it by [rerunning](https://jenkins.scylladb.com/job/manager-master/job/ubuntu24-upgrade-debug/3/) it with latest master which passed ok.
Then i added a fictional commit (doing nothing) and it [failed](https://argus.scylladb.com/tests/scylla-cluster-tests/ff42742f-b2fb-4235-9259-10d6fb7e1506) on these steps as well.
So i don't mind opening an SCT issue for that if you think it's important, but it should be outside of the scope of this PR.
Perhaps @fruch has any idea, if it's a jenkins + branch related issue.
cc: @pehala

pehala commented:
> @mikliapko , again, these errors are NOT related to this PR.
> i even confirmed it by [rerunning](https://jenkins.scylladb.com/job/manager-master/job/ubuntu24-upgrade-debug/3/) it with latest master which passed ok.
> Then i added a fictional commit (doing nothing) and it [failed](https://argus.scylladb.com/tests/scylla-cluster-tests/ff42742f-b2fb-4235-9259-10d6fb7e1506) on these steps as well.
> So i don't mind opening an SCT issue for that if you think it's important, but it should be outside of the scope of this PR.
> Perhaps @fruch has any idea, if it's a jenkins + branch related issue.

Do we have a successful rerun of that test if it is instability only?

yarongilor commented:
> > @mikliapko , again, these errors are NOT related to this PR.
> > i even confirmed it by [rerunning](https://jenkins.scylladb.com/job/manager-master/job/ubuntu24-upgrade-debug/3/) it with latest master which passed ok.
> > Then i added a fictional commit (doing nothing) and it [failed](https://argus.scylladb.com/tests/scylla-cluster-tests/ff42742f-b2fb-4235-9259-10d6fb7e1506) on these steps as well.
> > So i don't mind opening an SCT issue for that if you think it's important, but it should be outside of the scope of this PR.
> > Perhaps @fruch has any idea, if it's a jenkins + branch related issue.
> 
> Do we have a successful rerun of that test if it is instability only?

i don't think it's related to any instability since it passed with sct master branch and fail with all runs of private branches, including (as mentioned above) rerunning the same master head with an additional commit of:
```
commit 815b8f960e5a3d9d9d9d1d5931bf944c07deeb8c (HEAD -> test_test_id, origin/test_test_id)
Author: yarongilor <yarongilor@scylla.com>
Date:   Mon Jul 28 16:48:49 2025 +0300

    test(test_id): will it fail for test_id: No such file or directory?
    
    f:common.py       l:1990 c:utils                p:ERROR > test_id not found. Exit code: 1; Error details cat: /home/ubuntu/sct-results/latest/test_id: No such file or directory

diff --git a/mgmt_cli_test.py b/mgmt_cli_test.py
index 3b22db440..57e297d98 100644
--- a/mgmt_cli_test.py
+++ b/mgmt_cli_test.py
@@ -12,6 +12,7 @@
 # See LICENSE for more details.
 #
 # Copyright (c) 2016 ScyllaDB
+# Not doing any code change in this commit PR!!!
 

```

pehala commented:
> i don't think it's related to any instability since it passed with sct master branch and fail with all runs of private branches, including (as mentioned above) rerunning the same master head with an additional commit of:

Ok, please create issue for it then

yarongilor commented on sdcm/mgmt/operations.py:
```diff
@@ -0,0 +1,819 @@
+import yaml
+from invoke import exceptions
+
+from . import TaskStatus
+from .cli import ScyllaManagerTool
+from .. import mgmt
+from ..exceptions import FilesNotCorrupted
+from ..remote import shell_script_cmd, LOCALRUNNER
+from ..sct_events.system import InfoEvent
+from ..test_config import TestConfig
+from ..tester import ClusterTester
+from ..utils.azure_utils import AzureService
+from ..utils.cluster_tools import flush_nodes, major_compaction_nodes, clear_snapshot_nodes
+from ..utils.compaction_ops import CompactionOps
+from ..utils.gce_utils import get_gce_storage_client
+from ..utils.loader_utils import LoaderUtilsMixin
+from ..utils.time_utils import ExecutionTimer
```
The root cause is now fixed in [fix(logdir import): Move create-logging and log-directory to conftest.py](https://github.com/scylladb/scylla-cluster-tests/pull/11353/commits/f351549c6905d08ccb114a48d5f241faf9a1d709)

mikliapko replied:
I'm not fully following the original reason of using relative import but if the root cause is fixed, why do you still use relative imports here?

yarongilor commented:
> > i don't think it's related to any instability since it passed with sct master branch and fail with all runs of private branches, including (as mentioned above) rerunning the same master head with an additional commit of:
> 
> Ok, please create issue for it then

Opened https://github.com/scylladb/scylla-cluster-tests/issues/11546.
And fix in commit of [fix(logdir import): Move create-logging and log-directory to conftest.py](https://github.com/scylladb/scylla-cluster-tests/pull/11353/commits/f351549c6905d08ccb114a48d5f241faf9a1d709)

regevran commented on test-cases/performance/perf-regression-latency-backup-nemesis.yaml:
```diff
@@ -0,0 +1,49 @@
+# Divide by 10 (not 8) → 6250 / 10 = 625 MB/s
+# Convert to "disk MB/s" using 1000/1024: 625 * 1000 / 1024 = ~610.35 MB/s
+# Take 75% of that: 0.75 * 610.35 = ~457.76 MB/s
+append_scylla_args: '--stream_io_throughput_mb_per_sec 458 --stream-io-throughput-mb-per-sec 600 --blocked-reactor-notify-ms 5 --abort-on-lsa-bad-alloc 1 --abort-on-seastar-bad-alloc --abort-on-internal-error 1 --abort-on-ebadf 1'
```
You can keep `--stream_io_throughput_mb_per_sec 458` in `append-scylla_args` (but remove the second one with the value 600).
Or
Place the same value (458) in scylla.yaml by [appending](https://github.com/scylladb/scylla-cluster-tests/pull/11353/files/15719d5bcc4e81159eb948b98ef451b288b018a4#diff-7fb18ef7dfd6fda3a1d42c899ec2c544baf748af1766fb9560df35041facfd5aR15): `stream_io_throughput_mb_per_sec: 458`

pehala replied:
In that case, it would be not standard configuration as we do not expect customer to set this, as it is set autoamtically now, right?

regevran replied:
> In that case, it would be not standard configuration as we do not expect customer to set this, as it is set autoamtically now, right?

No. These is no "general" automation - in the cloud we set it as part of the installation. But just in the cloud.

yarongilor commented:
> [Retested](https://argus.scylladb.com/tests/scylla-cluster-tests/0432eed1-42b3-49ae-b71a-ca9c4c9bbcac) and runs ok after last addressed comment. @pehala , i think all comments are now addressed and tested, so PR is principally ready for merge if no other concerns. The one concern it has is that the PR of #11632 must first be merged.

update:
All comments addressed.
All commits are squashed.
[retested](https://argus.scylladb.com/tests/scylla-cluster-tests/787cc129-67af-4204-b630-661c50e49e82) and passed.

@pehala , @fruch , @mikliapko , @regevran , please consider approving if no further concerns (since the PR has 45 files and a high conflicts potential).

regevran commented:
>  @regevran , please consider approving if no further concerns 

Before I start my review - is there a chance to split into more review-consumable chunks of commits?
E.g. - introducing the datasets as a separate commit or whatever make sense?

pehala commented:
> @pehala , @fruch , @mikliapko , @regevran , please consider approving if no further concerns (since the PR has 45 files and a high conflicts potential).

Reason for high-conflict potential is https://github.com/scylladb/scylla-cluster-tests/pull/11353/commits/76b24bbae37d211c4852b2749b313f1d8383a375. If the conflicts or length of the PR causes trouble I would recomment extracting this into separate PR.

yarongilor commented:
[tested](https://argus.scylladb.com/tests/scylla-cluster-tests/d63f871f-26c8-484a-86e5-13ea269717a0) with 2025.3 rc2 as well
cc: @pehala

yarongilor commented:
@pehala , i think there's no point to resolve the same conflicts time after time when Mikita sends a new PR.
IMO, for such a wide PR there's no reason not to have a code freeze for manager code (at least mgmt_cli_test.py) until this refactor is merged.
Or alternatively we can freeze this PR until other Manager PRs are merged and the dependent PR of https://github.com/scylladb/scylla-cluster-tests/pull/11801

pehala commented:
> @pehala , i think there's no point to resolve the same conflicts time after time when Mikita sends a new PR.
> IMO, for such a wide PR there's no reason not to have a code freeze for manager code (at least mgmt_cli_test.py) until this refactor is merged.
> Or alternatively we can freeze this PR until other Manager PRs are merged and the dependent PR of #11801

No code freeze is needed, whiever will get in first wins, other has to rebase. Do the rebase now, there is no point waiting, it will be worse later. Review can be done in parallel on both PRs

yarongilor commented:
weekly ping @scylladb/qa-maintainers , @mikliapko - please review.

yarongilor commented:
> @yarongilor The part related to Manager refactoring - LGTM. The only thing I'd like to see is some fresh test runs for Manager to make sure we don't introduce any regression by these changes.

retested in:
[ubuntu24-upgrade](https://argus.scylladb.com/tests/scylla-cluster-tests/ed8692fb-6dae-4fbc-bc0a-37cb66d63da5)
[ubuntu24-installation](https://argus.scylladb.com/tests/scylla-cluster-tests/e39f9704-5aff-4cb3-b9f9-37fe97c47495)

vponomaryov commented on configurations/manager/850GB_backup_dataset.yaml:
```diff
@@ -0,0 +1,11 @@
+]
+
+# stress command to run
+stress_cmd_m: "cassandra-stress mixed cl=QUORUM duration=720m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate 'threads=125 fixed=20000/s' -col 'size=FIXED(1024) n=FIXED(1)'"
```
Is ti intentional that this command doesn't use `gauss` distribution as in in the `configurations/manager/2TB_backup_dataset.yaml` config?

yarongilor replied:
i think i didn't remove this file from the PR for the purpose of future internal debug retesting, so it's not critical to leave it that way. It's not the official results we should report.

vponomaryov commented on test-cases/performance/perf-regression-latency-backup-nemesis.yaml:
```diff
@@ -0,0 +1,49 @@
+email_subject_postfix: 'latency during operations'
+cluster_health_check: false
+stress_image:
+  cassandra-stress: 'scylladb/cassandra-stress:3.17.5'
```
In existing cases it is related to the `history` of results.
Here you add a new test.

So, if hardcode then hardcode current default.

yarongilor replied:
It's possible as well. Yet we also got a request of "maximum" alignment and standardized to existing perf tests.

yarongilor replied:
removed.

yarongilor commented on sdcm/utils/cluster_tools.py:
```diff
@@ -66,3 +69,24 @@ def check_cluster_layout(db_cluster: BaseCluster) -> bool:  # noqa: F821
+    LOGGER.debug("Run a flush on cluster data nodes")
+    triggers = [partial(node.run_nodetool, sub_cmd=f"flush -- {keyspace}", )
+                for node in cluster.data_nodes]
+    ParallelObject(objects=triggers, timeout=1200).call_objects()
```
it's also possible to add a TODO + followup issue in order to avoid retesting now.

yarongilor commented on sdcm/utils/cluster_tools.py:
```diff
@@ -66,3 +69,24 @@ def check_cluster_layout(db_cluster: BaseCluster) -> bool:  # noqa: F821
+    LOGGER.debug("Run a clear-snapshot command on cluster data nodes")
+    triggers = [partial(node.run_nodetool, sub_cmd="clearsnapshot", )
+                for node in cluster.data_nodes]
+    ParallelObject(objects=triggers, timeout=1200).call_objects()
```
Added a TODO and an issue: https://github.com/scylladb/scylla-cluster-tests/issues/11884

yarongilor commented on sdcm/utils/cluster_tools.py:
```diff
@@ -66,3 +69,24 @@ def check_cluster_layout(db_cluster: BaseCluster) -> bool:  # noqa: F821
+    LOGGER.debug("Run a major compaction on cluster data nodes")
+    triggers = [partial(node.run_nodetool, sub_cmd="compact", args=f"{keyspace} {table}", ) for
+                node in cluster.data_nodes]
+    ParallelObject(objects=triggers, timeout=3000).call_objects()
```
Added a TODO and an issue: https://github.com/scylladb/scylla-cluster-tests/issues/11884

vponomaryov reviewed (approved):
LGTM

mikliapko reviewed (approved):
LGTM
